### ISSUE TITLE
import mill project on MacOS 10.15 with Metals 0.8.0 doesn't work

### ISSUE THREAD

**Issue by tylerprete** [scalameta/metals#1298]
**Describe the bug**
When attempting to import a simple mill project with metals 0.8.0 in vscode, I'm getting "import project failed, no functionality will work."

Appears to be attempting to write to /.metals/download instead of my home directory, which is odd, because in metals/src/main/resources/millw the download path is set like so:
```
MILL_DOWNLOAD_PATH="${HOME}/.mill/download"
```

Perhaps it can't pick up $HOME in the context it's running in?

**To Reproduce**
1. Download my mill project (or create one): https://github.com/tylerprete/little-mill
2. Attempt vscode/metals import from cmdline with ```code .``` in checkout folder.
3. Run "Metals: Import Build" manually if VSCode doesn't suggest it automatically
3. See error

**Expected behavior**
Metals is able to download mill successfully and setup bloop.

**Screenshots**

**Installation:**
- Operating system: macOS Catalina (10.15)
- Editor: Visual Studio Code
- Metals version: v0.8.0

**Additional context**
metals.log contain this:
```
INFO  started: Metals version 0.8.0 in workspace '/Users/tprete/repos/little-mill'
INFO  time: initialize in 0.32s
INFO  running '/var/folders/tp/p9cbzcjj5nb268sl3bnth19w0000gn/T/metals5540375500728896923/millw --mill-version 0.5.2 --predef /var/folders/tp/p9cbzcjj5nb268sl3bnth19w0000gn/T/metals5540375500728896923/predef.sc mill.contrib.Bloop/install'
INFO  Downloading mill 0.5.2 from https://github.com/lihaoyi/mill/releases ...
INFO    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
INFO                                   Dload  Upload   Total   Spent    Left  Speed
100   605    0   605    0     0   1563      0 --:--:-- --:--:-- --:--:--  1563
100 49.9M  100 49.9M    0     0  22.1M      0  0:00:02  0:00:02 --:--:-- 41.9M
INFO  mkdir: /.mill/download: Read-only file system
INFO  build tool exit: 1
INFO  time: ran 'mill bloopInstall' in 2.48s
ERROR Mill command failed: /var/folders/tp/p9cbzcjj5nb268sl3bnth19w0000gn/T/metals5540375500728896923/millw --mill-version 0.5.2 --predef /var/folders/tp/p9cbzcjj5nb268sl3bnth19w0000gn/T/metals5540375500728896923/predef.sc mill.contrib.Bloop/install
INFO  shutting down Metals
```

**Search terms**
failed mill project import

**Comment by gabro** [comment]
Thanks for reporting!
How are you launching VSCode?
Environment variables may not get picked up when launching from the app outside the terminal.
If that’s the case, could you try running `code /path/to/project` a terminal and reproduce?

We should definitely check the HOME variable and provide a meaningful message, anyway.

**Comment by gabro** [comment]
Ah, nvm. I saw now that you’re using the terminal. I’ll try to reproduce tomorrow, thanks for now

**Comment by ckipp01** [comment]
So I was actually able to reproduce this using VS Code both when launching from the terminal and when just opening the app. However, when I try the same project with vim, it works find 🤔

**Comment by jodersky** [comment]
I can reproduce this too on Linux, Debian.

I added an "env" statement in the millw script [here](https://github.com/scalameta/metals/blob/ebd00ad7b62d452fed31a8275242cc9cdb13aaaf/metals/src/main/resources/millw#L42-L43) and published locally. The results indeed confirm that:

1. when "importing the build" through vscode, $HOME is not set. In fact, only the environment variables [defined here](https://github.com/scalameta/metals/blob/5054a085c03e78b119d7ba0df2efafaefb0f8675/metals/src/main/scala/scala/meta/internal/metals/BloopInstall.scala#L78) are available.
2. When manually running `millw --mill-version xxx --predef /... mill.contrib.Bloop/install` *from my regular shell*, all environment variables are available, including HOME.

In conclusion, it appears that VSCode (or whatever mechanism metals is invoked from in vscode) does not forward the user's environment.

**Comment by jodersky** [comment]
I continued investigating this, and here is my suspicion.

1. About a month ago, a PR was merged to metals-vscode that allows setting coursier repositories https://github.com/scalameta/metals-vscode/pull/161.
2. A commit in this PR sets an "env" parameter in the language ServerOptions  https://github.com/scalameta/metals-vscode/pull/161/files#diff-45327f86d4438556066de133327f4ca2L256
3. According to the [process spawning utilities of vscode](https://github.com/microsoft/vscode/blob/ddd47f98c890a3e54f81aa0c236dca9753cf8acb/src/vs/base/common/processes.ts#L18-L22), the "env" parameter represents the following:

> The environment of the executed program or shell. **If omitted
> the parent process' environment is used**.

(emphasis mine)

My suspicion is that in setting the environment flag, the user's default env no longer gets forwarded, hence leading to the issue discussed here.

Note that I haven't tested this yet, so it is only a suspicion.

**Comment by olafurpg** [comment]
@jodersky thank you for investigating. That sounds like a likely cause of this issue. The metals extension should make sure to pass along the parent process environment
